### PR TITLE
Transfer improvements

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

**Comment by fluffypony** [comment]
Please rebase

File: src/wallet/wallet2.h [hunk 1]
```diff
@@ -116,7 +117,7 @@ namespace tools
     //! Uses stdin and stdout. Returns a wallet2 and password for wallet with no file if no errors.
     static std::pair<std::unique_ptr<wallet2>, password_container> make_new(const boost::program_options::variables_map& vm);
 
-    wallet2(bool testnet = false, bool restricted = false) : m_run(true), m_callback(0), m_testnet(testnet), m_always_confirm_transfers(true), m_store_tx_info(true), m_default_mixin(0), m_default_priority(0), m_refresh_type(RefreshOptimizeCoinbase), m_auto_refresh(true), m_refresh_from_block_height(0), m_confirm_missing_payment_id(true), m_restricted(restricted), is_old_file_format(false) {}
+    wallet2(bool testnet = false, bool restricted = false) : m_run(true), m_callback(0), m_testnet(testnet), m_always_confirm_transfers(true), m_store_tx_info(true), m_default_mixin(0), m_default_priority(0), m_refresh_type(RefreshOptimizeCoinbase), m_auto_refresh(true), m_refresh_from_block_height(0), m_confirm_missing_payment_id(true), m_restricted(restricted), is_old_file_format(false), m_node_rpc_proxy(m_http_client, m_daemon_rpc_mutex) {}
```
Comment (NanoAkron) [inline src/wallet/wallet2.h]: Please consider putting before m_restricted and is_old_file_format to keep the same order in both lists

File: src/wallet/wallet_rpc_server.cpp [hunk 2]
```diff
@@ -394,6 +394,7 @@ namespace tools
     std::vector<cryptonote::tx_destination_entry> dsts;
     std::vector<uint8_t> extra;
 
+    LOG_PRINT_L3("on_transfer_split starts");
```
Comment (NanoAkron) [inline src/wallet/wallet_rpc_server.cpp]: Use your new log levels?

File: src/wallet/wallet2.h (hunk 1 above)
Comment (moneromooo-monero) [inline src/wallet/wallet2.h]: No, it's meant to be in declaration order. If a line is missing some, they should be added instead (assuming they're needed there).

File: src/wallet/wallet_rpc_server.cpp (hunk 2 above)
Comment (moneromooo-monero) [inline src/wallet/wallet_rpc_server.cpp]: When they're merged

**Comment by moneromooo-monero** [comment]
rebase done

**Pull request merged**